staging/src/k8s.io/client-go/tools/events/event_recorder.go

```diff
@@ -68,7 +68,7 @@ func (recorder *recorderImpl) makeEvent(refRegarding *v1.ObjectReference, refRel
 	t := metav1.Time{Time: recorder.clock.Now()}
 	namespace := refRegarding.Namespace
 	if namespace == "" {
-		namespace = metav1.NamespaceSystem
```

Comment:
Are any in-tree components affected by this change? If so, at a minimum the release note needs to list them, but more likely we'd want to not make this change or make the reverse change for each user. People looking for events could be unexpectedly broken by this. And the default change depends on your version of client go, which might confuse users of client go.

What is the benefit of this change?

@liggitt @deads2k

Reply:
I agree the two recorders should behave the same. Components fall back from this one (using the v1beta1 API) to the original one (using the core v1 API) if the v1beta1 API is not available. Defaulting namespaces differently makes event recorder behavior unpredictable.

I would tend to preserve the legacy core v1 behavior

Reply:
+1 for Jordan

Reply:
... now I'm super-confused ... we've hit mismatches between the event reporter and API validation in https://github.com/kubernetes/kubernetes/issues/100119

an integration test run against 1.19-1.20 shows that API validation prevents creating events for a cluster-scoped object in the default namespace... https://github.com/kubernetes/kubernetes/pull/100125/files#r702165538